### PR TITLE
login: require CSS flex & grid support

### PR DESCRIPTION
**Pull request by garrett**
- [x] Depends on #10568.

### CODE REVIEW

**Comment by garrett** [comment]
We still should decide if we want to support browsers that support supports but not support grid. The website PR https://github.com/cockpit-project/cockpit-project.github.io/pull/221 implements the version bump. (https://github.com/cockpit-project/cockpit-project.github.io/pull/221/files is the diff of versions. Minimum version is between mid to end of 2017, depending on browser.)

Pro for requiring grid support: We wouldn't have to worry about and implement fallbacks for browsers in a small window between supporting supports and not supporting grid yet.

Contra for mandatory grid support: fewer browsers supported. But, realistically, the unsupported browsers are between a year to a year and a half old at this point. People should be using newer browsers for some time now, between Windows service packs bumping up Edge support, Safari getting updated in iOS/macOS, and Firefox ESR supporting grid for quite a while now. (Chrome and others auto-update, so they're non-issues.)

In addition to grid, I added flex too, but all browsers have supported flex for years.

**Comment by garrett** [comment]
This will probably fail tests until #10568 is merged, as it calls a function that's added in the other PR.

**Comment by martinpitt** [comment]
This will bump the required versions on https://cockpit-project.org/running.html, right?

What's the impending use case for introducing grids? Is that for the VM storage pool work?

**Comment by martinpitt** [comment]
#10568 landed.

**Comment by garrett** [comment]
> This will bump the required versions on https://cockpit-project.org/running.html, right?

Yep, already submitted a PR yesterday @ https://github.com/cockpit-project/cockpit-project.github.io/pull/221 :wink:

**Comment by martinpitt** [comment]
Semaphore fails:
```
./tools/missing ./tools/jshint src/ws/login.js
src/ws/login.js:160:16: 'css' is not defined.
src/ws/login.js:161:16: 'css' is not defined.
```

**Comment by martinpitt** [comment]
This seems to break the Edge tests. Maybe our Win 10 image is too old?

**Comment by martinpitt** [comment]
@Gundersanne, are you up for refreshing our win 10 images? I hope MS updates the images regularly, or we have to install pending updates as part of image creation?

**Comment by martinpitt** [comment]
Our Edge isn't actually that old: Edge 42.17, EdgeHTML 17.17

So it just doesn't seem to like the `display: flex` check.

**Comment by garrett** [comment]
Apparently, Edge does not properly support the CSS supports JavaScript API. None of the docs except MDN (and just recently) mention this. And it's also not mentioned anywhere else on the Internet. Fun.

Here's where MDN mentions it, as a footnote, when you expand the Edge support:
https://developer.mozilla.org/en-US/docs/Web/API/CSS/supports

(The MDN docs in devdocs.io don't even mention this, so it's a very recent change.)

Accepted by all modern browsers:
```js
CSS.supports("display", "inline");
```
```js
CSS.supports("(display: inline)");
```

Supported by every modern browser except Edge:
```js
CSS.supports("display: inline");
```

I've reworked the patch to check in the first way `("foo", "bar")`, as that's similar to what we currently have. As we might want to also check for a specific CSS string instead of a key, value pair in the future (as strings are more flexible), I've also commented this issue in the local `css()` block.

---

_**TL;DR**: Edge edge case in supports support._
